src/ray/raylet/node_manager.cc

```diff
@@ -618,10 +698,19 @@ void NodeManager::ProcessClientMessage(
       RAY_CHECK(!driver_id.is_nil());
       local_queues_.RemoveDriverTaskId(driver_id);
       worker_pool_.DisconnectDriver(driver);
+
+      RAY_LOG(DEBUG) << "Driver (pid=" << driver->Pid() << ") is disconnected. "
+                     << "driver_id: " << driver->GetAssignedDriverId();
     }
     return;
   } break;
   case protocol::MessageType::SubmitTask: {
+    std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
+    if (worker && worker->IsDead()) {
+      // break out if the worker is marked as dead. This can happen
+      // when a worker is being killed.
+      break;
+    }
```

Comment:
this check is repeated many times, should we put it inside `GetRegisteredWorker` (maybe add a flag `includeDead=false` if needed)?

Reply:
Actually for the dead worker, the only allowed message is `DisconnectClient`, it's probably simpler to just put this check at the beginning of `ProcessClientMessage`.